src/services/validation.js

```diff
@@ -0,0 +1,22 @@
+import firebaseAPI from "@/services/API/checkUser.js";
+import { sendPlio } from "./sendPlio";
+
+export async function validateUser(userID, validateCount, isSingleEntryOnly, redirectID, doesUserExist){
+    //response tells us if the user is authenticated.
+    if(!(/^([0-9])\1*$/.test(userID)) && userID.toString()[0] == '1'){
+        doesUserExist = await firebaseAPI.checkUserExists(userID);
+    }
+    // this condition checks if the user is getting authenticated the first time. Just shows an error message.
+    if (!doesUserExist && validateCount == 0) {
+    validateCount = 1;
+    var invalidLoginMessage = "Please enter correct SRN / कृपया सही SRN दर्ज करें";
```

Comment:
because this variable is declared in the if block, if the compiler doesn't reach this if condition, this will be undefined when you're returning it at the end. 
You can declare the variable outside the if block, that way it will be accessible everywhere